frontend/src/App.tsx

```diff
@@ -334,42 +367,67 @@ const Form = (props: { model: Model, dispatch: React.Dispatch<Msg> }) => {
 export const EntrypointForms = (props: { model: Model, dispatch: React.Dispatch<Msg> }) => {
 
   const isDisabled = (): string => {
-    if (props.model.contract._tag === "RemoteSuccess") return 'opacity-100 pointer-events-auto'
-    else return 'opacity-50 pointer-events-none'
+    if (props.model.contract._tag === "RemoteSuccess") return ' opacity-100 pointer-events-auto'
+    else return ' opacity-50 pointer-events-none'
   }
 
   const allEntrypoints: Array<[string, EntrypointTypes]> = [
     ["Swap X for Y", { type: 'x_to_y' }],
     ["Swap Y for X", { type: 'y_to_x' }],
     ["Set position", { type: 'set_position' }],
+    ["Update position", { type: 'update_position' }],
     ["Transfer position", { type: 'transfer' }],
     ["Update operators", { type: 'update_operators' }],
     ["Swap X for X'", { type: 'x_to_x_prime' }],
     ["Increase observation count", { type: 'increase_observation_count' }],
   ]
 
-  let ErrorText = () => {
+  let ResultText = () => {
     if (props.model.callEntrypointResult._tag === "RemoteFailure")
       return <div className="py-2">
         <p className="text-sm text-red-600">{JSON.stringify(props.model.callEntrypointResult.error)}</p>
       </div>
+    else if (props.model.callEntrypointResult._tag === "RemoteSuccess")
+      return <div className="py-2">
+        <p className="text-sm text-green-600">{JSON.stringify(props.model.callEntrypointResult.value)}</p>
+      </div>
     else return <div></div>
   }
 
-  return <div className={"bg-white rounded-2xl mt-10 m-auto " + isDisabled()} style={{ width: "500px" }}>
+  let ContractStorageView = () => {
+    if (props.model.storage._tag === "RemoteSuccess")
+      return <div>
+        <p className="pb-1">Sqrt price: {props.model.storage.value.sqrt_price.toString()}</p>
+      </div>
+    else if (props.model.storage._tag === "RemotePending")
+      return <div>Loading...</div>
+    else return <div></div>
+  }
 
-    <div className="relative">
-      <div className="absolute top-0 bg-white py-4 rounded-lg" style={{ "left": "-260px" }}>
-        {pipe(allEntrypoints, A.mapWithIndex((i, a) => <FormSelect key={i} a={a} dispatch={props.dispatch} model={props.model} />))}
+  return <div>
+    <div className={"bg-white rounded-2xl mt-10 m-auto p-6 " + isDisabled()} style={{ width: "500px" }}>
+      <p className="font-bold pb-1">Relevant Contract Storage</p>
```

Comment:
This seems a bit of an odd name, I don't think users care about the where that info came from.

I'd rephrase this as  "current contract info".

Also, what do you think about adding a button to manually refresh this?